examples/tensorflow-profiling-examples/example_test.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import testing_utils
+from tensorflow.python.keras import initializers
+
+import example
+
+
+class LinearBlockFullTest(tf.test.TestCase):
+    def test_basic(self):
+       testing_utils.layer_test(example.LinearBlockFull, input_shape=(4, 32))
+
+    def test_output(self):
+        batch_size, dim, output_dim = (3, 4, 2)
+        testing_utils.layer_test(
+            example.LinearBlockFull,
+            kwargs={'units': output_dim},
+            input_data=np.ones((batch_size, dim)),
+            expected_output_dtype='float32'
+        )
+
+
+class LinearBlockTest(tf.test.TestCase):
+    def test_shape_default(self):
+        x = np.ones((4, 32))
+        layer = example.LinearBlock()
+        output = layer(x)
+        self.assertAllEqual(output.shape, (4, 32))
+
+    def test_shape(self):
+        batch_size, dim = (4, 15)
+        x = np.ones((batch_size, dim))
+        layer = example.LinearBlock(dim)
+        output = layer(x)
+        self.assertAllEqual(output.shape, (batch_size, dim))
+
+    def test_shape_multidim(self):
+        batch_size, dim = (4, 15)
+        x = np.ones((batch_size, 7, dim))
+        layer = example.LinearBlock(dim)
+        output = layer(x)
+        self.assertAllEqual(output.shape, (batch_size, 7, dim))
+
+    def test_output(self):
+        dim = 4
+        batch_size = 3
+        x = np.ones((batch_size, dim))
+        layer = example.LinearBlock(dim)
+        output = layer(x)
+        expected_output = np.array([[0.1250, -0.0576, 0.0513, -0.0305]]*batch_size)
+        self.assertAllClose(output, expected_output, atol=1e-4)
+
+    @patch.object(initializers, 'get', lambda _: tf.compat.v1.keras.initializers.Ones)
+    def test_output_ones(self):
+        dim = 4
+        batch_size = 3
+        output_dim = 2
+        x = np.ones((batch_size, dim))
+        layer = example.LinearBlock(output_dim)
+        output = layer(x)
+        expected_output = np.ones((batch_size, output_dim)) * (dim + 1)
```

Comment:
nit: this might need a comment or docstring explaining how it's essentially `np.ones((batch_size, dim)).dot(np.ones((dim, output_dim))) + np.ones((batch_size, 1))` because the initializes for `w` and `b` were replaced with ones.